File: src/old/app/functions/Parser.java
package app.functions;

import java.util.function.Consumer;
/**
 * @author Jack Meng
 */
public abstract class Parser {
  public static <T> void foreach(T[] arr, Consumer<T> consumer) {
    for (T t : arr) {
      consumer.accept(t);
    }
  }
}

File: kt/app/functions/Parser.kt
package app.functions

import java.lang.Runnable
import java.util.function.Consumer
import kotlin.jvm.Synchronized

/**
 * @author Jack Meng
 */
object Parser {
    @JvmStatic
    fun <T> foreach(arr: Array<T>, consumer: Consumer<T>) {
        for (t in arr) {
            consumer.accept(t)
        }
    }
}